packages/react/src/hooks/usePreservedCallback.ts

```diff
@@ -0,0 +1,14 @@
+import { useCallback, useRef } from 'react'
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect'
+
+export const usePreservedCallback = <T extends (...args: unknown[]) => unknown>(callback: T) => {
+  const callbackRef = useRef<T>(callback)
+
+  useIsomorphicLayoutEffect(() => {
+    callbackRef.current = callback
+  }, [callback])
+
+  return useCallback((...args: unknown[]) => {
+    return callbackRef.current(...args)
+  }, []) as T
+}
```

Comment:
This is slightly different from `toss/slash's usePreservedCallback`. 

I used `useIsomorphicLayoutEffect` instead of `useEffect`, and changed the type to `unknown` instead of `any`.